File: Maths/ChkPalindrome.py
def is_palindrome(x):
    rev = 0
    temp = x
    while x != 0:
        ld = x % 10 
        rev = rev * 10 + ld
        x //= 10
    return temp == rev

for _ in range(2):
    x = int(input("Enter the nummber: "))
    print("The result is: ",bool(is_palindrome(x)))

File: C/Day3/Recursion/ChkPalindrome.c
#include <stdio.h>
#include <stdbool.h>
int rev(int n, int temp);
int chk(int n);
int main(){
    printf("%s",chk(1010) ? "true" : "false");
    return 0;
}
int rev(int n, int temp) {
    if (n == 0) {
        return temp;
    }
    if (n < 0) {
        n = -n;
    }
    temp = (temp * 10) + (n % 10);
    return rev(n/10, temp);
}
int chk(int n){
    if(n == rev(n,0)){
        return 1;
    }else{
        return 0;
    }
}